src/workloads/query/TelemetryWithRedaction.yml

```diff
@@ -0,0 +1,234 @@
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            # This will create a find command with a random number of fields from 3 to 350, to create varying query shapes and sizes.
+            Filter:
+              {$and:
+                [{a1: 1}, {a2: 1}, {^Object: {
+                  withNEntries: {^RandomInt: {min: 1, max: 350}},
+                  havingKeys: {^RandomString: {length: 10}},
+                  andValues: {$exists: false},
+                  duplicatedKeys: skip
+                }}]
+              }
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [5]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- Name: FindConcurrentlyWithTelemetryWithRedaction
+  Type: CrudActor
+  Threads: 32
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [6]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter:
+              {$and:
+                [{a1: 1}, {a2: 1}, {^Object: {
+                  withNEntries: {^RandomInt: {min: 1, max: 350}},
+                  havingKeys: {^RandomString: {length: 10}},
+                  andValues: {$exists: false},
+                  duplicatedKeys: skip
+                }}]
+              }
+
+# Telemetry with redaction.
+- ActorFromTemplate:
+    TemplateName: TelemetryAggregation
+    TemplateParameters:
+      Name: "TelemetryAggregationWithRedaction"
+      applyHmacToIdentifiers: true
+      OnlyActiveInPhase: 6
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [7]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- Name: FindConcurrentlyWithTelemetryWithoutRedaction
+  Type: CrudActor
+  Threads: 32
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [8]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter:
+              {$and:
+                [{a1: 1}, {a2: 1}, {^Object: {
+                  withNEntries: {^RandomInt: {min: 1, max: 350}},
+                  havingKeys: {^RandomString: {length: 10}},
+                  andValues: {$exists: false},
+                  duplicatedKeys: skip
+                }}]
+              }
+
+# Telemetry without redaction.
+- ActorFromTemplate:
+    TemplateName: TelemetryAggregation
+    TemplateParameters:
+      Name: "TelemetryAggregationWithoutRedaction"
+      applyHmacToIdentifiers: false
+      OnlyActiveInPhase: 8
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone-all-feature-flags
```

Comment:
Can you add regular 'standalone' here too?

Reply:
Oh actually nvm - that won't work, right? Can't run $telemetry without the feature flag.

Reply:
So we should explicitly compare the two you list here in the perf comparison report - the 'all-feature-flags' will have the rate limit set to 0 and then we can see the impact of collecting telemetry also.

Reply:
That sounds good! Should I add it to `scripts/perf_reports_query_telemetry_perf.yml` as part of this ticket?

Reply:
Yes please! I can talk you through how to test it out. Let's merge this first though and give it a chance to run so we have some data to test out